packages/good-design/src/core/inputs/WebVideoUploader.tsx

```diff
@@ -0,0 +1,44 @@
+import { View } from "native-base";
+import React, { useCallback, useRef } from "react";
+import { GoodButton } from "..";
+
+type Props = {
+  onDone: (video?: { base64: string; extension: string }, error?: Error) => Promise<void>;
+};
+
+export const WebVideoUploader = ({ onUpload, isLoading }: { onUpload: Props["onDone"]; isLoading: boolean }) => {
+  const fileRef = useRef<any>(null);
+
+  const handleVideoUpload = useCallback(
+    (event: any) => {
+      event.preventDefault();
+      const imageFile = event.target.files[0];
+      if (imageFile) {
+        console.log({ imageFile });
+        const reader = new FileReader();
+        reader.onload = function (r) {
+          const extension = imageFile.name.match(/\.([^.]+)$/)?.[1] || "mp4";
+
+          void onUpload({ base64: ((r.target?.result as string) || "").split(",")[1] as string, extension });
```

Comment:
don't we have to validate file by extension & content-type ?

Reply:
yes we should. added file extension and file-size validation here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/148/commits/5d185b2ec1e692d5971cbed764ac0751f2094107